contents/blog/measuring-feature-success.md

```diff
@@ -0,0 +1,114 @@
+
+This is exactly what Yakko did with sampling. He picked one insight (lifecycle was easiest), updated the API, built a quick UI, and [shipped it](https://github.com/PostHog/posthog/pull/14283) behind a feature flag. He moved quickly to test in production with production data. 
+
+![Lifecycle](../images/blog/measuring-feature-success/lifecycle.png)
+
+A feature is successful at this stage if it solves the issue, isn’t broken, and can scale to millions of events and users. Since we use PostHog at PostHog, this often means using it ourselves (aka dogfooding). Feature flags let us roll out or roll back quickly if needed.
+
+### 2. Beta test and get feedback
+
+Once an MVP of a feature ships, it continues the process of evaluation by rolling out further. More members of our team try out the feature and provide feedback.
+
+For sampling, Yakko added more insights that supported sampling and then started polishing. He asked the team to try out the feature, wrote [docs](/docs/product-analytics/sampling), and had a chat with our Head of Product Luke about what sampling should look and feel like. He also started to take Michael's UI feedback more seriously.
+
+If a specific user requested the feature or has a problem potentially solved by it, we might roll it out to them with the expectation that it is recently built. Feature flags enable us to adjust the rollout and do this fast. We communicate with the user through this process and ideally receive feedback that helps us improve it.
+
+Sampling got its first real users through the support hero tagging Yakko in places they felt sampling might help out. This helped improve the experience for customers without completing the feature. When these users were positive about their experience, he knew the feature was successful so far and continued the rollout. 
+
+![Message](../images/blog/measuring-feature-success/message.png)
+
+A feature is successful at this stage if users use it and the qualitative feedback we receive from them is positive. A common goal at this stage is "5 happy teams using it."
+
+### 3. Launch
+
+When we have positive feedback and usage in beta, and the feature reaches the point we are ready to release. The flag rolls out further, eventually to 100% of users.
+
+Yakko continued to update the simple UI for sampling (it changed four times) to ultimately land on something he was happy with. He instrumented tracking for the feature’s usage and rolled it out to 25% of all organizations in PostHog.
+
+![Sampling UI](../images/blog/measuring-feature-success/sampling.png)
+
+At this stage, feature success is broad usage. A significant portion of target users should be using the feature. What this portion is depends on the usage of similar features. The feature should also be issue free and work as intended.
```

Comment:
> A significant portion of target users should be using the feature.

I think this is a really meaty topic to dig into and ultimately the difficult part of defining feature success. How do I know how much adoption is good? Is 5% of users using it good? 10%? 50? How do we determine this

Reply:
I tried to add some metrics based on our numbers, but they range so much 😅